test/ZkMinterModTriggerV1.t.sol

```diff
@@ -1,343 +1,350 @@
-    balanceOf[to] += amount;
-    return true;
+  function test_InitializesMinterTriggerCorrectly() public view {
+    assertTrue(minterTrigger.hasRole(minterTrigger.DEFAULT_ADMIN_ROLE(), admin));
+    assertTrue(minterTrigger.hasRole(minterTrigger.PAUSER_ROLE(), admin));
+    assertEq(address(minterTrigger.mintable()), address(mintable));
+    assertEq(minterTrigger.targets(0), address(mockTarget));
+    assertEq(minterTrigger.calldatas(0), abi.encodeWithSelector(mockTarget.setValue.selector, 42));
+    assertEq(minterTrigger.values(0), 100 ether);
   }
 }
 
-contract ZkMinterModTriggerV1Test is Test {
-  ZkMinterModTriggerV1 public trigger;
-  ZkMinterModTargetExampleV1 public target;
-  MockERC20 public token;
-  address public user = address(0x123);
-
-  ZkCappedMinterV2 public cappedMinter;
-  address cappedMinterAdmin = makeAddr("cappedMinterAdmin");
-  bytes32 constant MINTER_ROLE = keccak256("MINTER_ROLE");
-
-  event TransferProcessed(address indexed sender, uint256 amount);
-
-  function setUp() public virtual {
-    token = new MockERC20();
-    target = new ZkMinterModTargetExampleV1(address(token));
-
-    // Prepare arrays for constructor: two calls
-    // 1. Approve on token contract
-    // 2. executeTransferAndLogic on target contract
-    address[] memory targetAddresses = new address[](2);
-    targetAddresses[0] = address(token); // Token contract for approve
-    targetAddresses[1] = address(target); // Target contract for executeTransferAndLogic
-
-    bytes[] memory functionSignatures = new bytes[](2);
-    functionSignatures[0] = abi.encodeWithSignature("approve(address,uint256)");
-    functionSignatures[1] = abi.encodeWithSignature("executeTransferAndLogic(uint256)");
-
-    bytes[] memory callDatas = new bytes[](2);
-    callDatas[0] = abi.encode(address(target), uint256(500 ether)); // Approve target to spend 500 ether
-    callDatas[1] = abi.encode(uint256(500 ether)); // Execute transfer of 500 ether
-
-    // Deploy ZkMinterModTriggerV1 with arrays
-    trigger = new ZkMinterModTriggerV1(cappedMinterAdmin, targetAddresses, functionSignatures, callDatas);
-
-    // Setup ZkCappedMinterV2
-    uint48 startTime = uint48(block.timestamp);
-    uint48 expirationTime = uint48(startTime + 3 days);
-    uint256 cap = 500e18;
-
-    cappedMinter = new ZkCappedMinterV2(
-      IMintable(address(token)), // Assuming MockERC20 is compatible with IMintable
-      cappedMinterAdmin,
-      cap,
-      startTime,
-      expirationTime
-    );
-    vm.prank(cappedMinterAdmin);
-    trigger.setMinter(address(cappedMinter));
-    vm.prank(cappedMinterAdmin);
-    cappedMinter.grantRole(MINTER_ROLE, address(trigger));
+contract Constructor is ZkMinterModTriggerV1Test {
+  function testFuzz_InitializesMinterTriggerCorrectly(
+    IMintable _mintable,
+    address _admin,
+    uint256 _setValue,
+    uint256 _value
+  ) public {
+    vm.assume(_admin != address(0) && address(_mintable) != address(0));
```

Comment:
Can mintable now be the 0 address?

Reply:
Yes! removed the assumes. https://github.com/zksync-association/zkminters/pull/33/commits/256cdffe247d62f21cf814cea469f928298914e2